quaistats/quaistats.go

```diff
@@ -168,6 +172,10 @@ func (q *StatsQueue) Enqueue(item interface{}) {
 	q.mutex.Lock()
 	defer q.mutex.Unlock()
 
+	if len(q.data) >= c_maxQueueSize {
```

Comment:
This turns the queue into a lossy cache. Are the algos which depend on queued data tolerant to this?